tests/analyses/new/test_systemviewers.py

```diff
@@ -0,0 +1,216 @@
+
+        with expectation:
+            result = viewer._validate_wavelength(wavelength)  # noqa: SLF001
+            assert result == expected
+
+    @pytest.mark.parametrize(
+        "field,expected,expectation",
+        [
+            (1, 1, does_not_raise()),
+            (4, 4, does_not_raise()),
+            (-1, -1, does_not_raise()),
+            ("All", -1, does_not_raise()),
+            ("a", None, pytest.raises(ValueError, match="field must be an integer or 'All'")),
+            (0.55, None, pytest.raises(TypeError, match="field must be an integer or 'All'")),
+            (
+                -2,
+                None,
+                pytest.raises(ValueError, match="field must be -1 or between 1 and the number of fields"),
+            ),
+            (0, None, pytest.raises(ValueError, match="field must be -1 or between 1 and the number of fields")),
+            (
+                5,
+                None,
+                pytest.raises(ValueError, match="field must be -1 or between 1 and the number of fields"),
+            ),
+        ],
+    )
+    def test_validate_field(self, field, expected, expectation, simple_system):
+        while simple_system.SystemData.Fields.NumberOfFields < 4:
+            simple_system.SystemData.Fields.AddField(1, 1, 1)
+
+        viewer = TestBase.MockSystemViewer()
+        viewer.run(simple_system)
+
+        with expectation:
+            result = viewer._validate_field(field)  # noqa: SLF001
+            assert result == expected
+
+    @pytest.mark.parametrize(
+        "start_surface,end_surface,expected,expectation",
+        [
+            (1, 2, 2, does_not_raise()),
+            (1, -1, 4, does_not_raise()),
+            (
+                1,
+                5,
+                None,
+                pytest.raises(
+                    ValueError,
+                    match="end_surface must be -1 or greater than start_surface and less than the number of surfaces",
+                ),
+            ),
+            (
+                3,
+                1,
+                None,
+                pytest.raises(
+                    ValueError,
+                    match="end_surface must be -1 or greater than start_surface and less than the number of surfaces",
+                ),
+            ),
+        ],
+    )
+    def test_validate_end_surface(self, start_surface, end_surface, expected, expectation, simple_system):
+        viewer = TestBase.MockSystemViewer()
+        viewer.run(simple_system)
+
+        with expectation:
+            result = viewer._validate_end_surface(start_surface, end_surface)  # noqa: SLF001
+            assert result == expected
+
+    @pytest.mark.skip_for_opticstudio_versions(">=24.1.0", "Settings are supported from OpticStudio 24R1")
+    def test_warn_ignored_settings(self):
+        viewer = TestBase.MockSystemViewer(number=6)
+
+        with pytest.warns(UserWarning, match="Some parameters were specified but ignored"):
+            viewer.run()
```

Comment:
@crnh it currently does not pass as `viewer.run()` requires an OpticStudioSystem, something like this should work:

```suggestion
    def test_warn_ignored_settings(self, simple_system):
        viewer = TestBase.MockSystemViewer(number=6)

        with pytest.warns(UserWarning, match="Some parameters were specified but ignored"):
            viewer.run(simple_system)
```